### PR TITLE
Editor release 1.42.2 integration

### LINKED ISSUE #2886: Crash when deleting block from page template
**Describe the bug**
When I create a page using a template and then delete a block, the app crashes - and not very gracefully. 

**To Reproduce**
1. In the app, create a new page using a template for a site with Jetpack installed. - I can only repro this on JP sites so far 
2. In the editor, delete a pre-set block
Crash
https://cloudup.com/ctvbVn5iqHn

**Expected behavior**
Block deletes with no crash

**Screenshots**
https://cloudup.com/cN4cvHBFV0e

**Smartphone (please complete the following information):**
iPhone 12 Pro, iOS 14.2.1, WP 16.3 beta

**Additional context**

### PR DESCRIPTION
**Pull request by guarani**
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/2886

`Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/2893
`Gutenberg PR` -> https://github.com/WordPress/gutenberg/pull/27683

To test check the `Gutenberg` PR description.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jd-alexander** [comment]
@guarani shouldn't this point to the `release/16.3` branch?

**Comment by guarani** [comment]
> @guarani shouldn't this point to the `release/16.3` branch?

True! Thanks for noticing!

**Comment by guarani** [comment]
Tested build 39279 and https://github.com/wordpress-mobile/gutenberg-mobile/issues/2886 is resolved 👍

File: Podfile [hunk 1]
```diff
@@ -154,7 +154,7 @@ target 'WordPress' do
     ## Gutenberg (React Native)
     ## =====================
     ##
-    gutenberg :tag => 'v1.42.0'
+    gutenberg :tag => 'v1.42.2'
```
Comment (jkmassel) [inline Podfile:157]: I see this is changing by two versions – just want to ensure that's deliberate? 😃

Comment (cameronvoell) [reply Podfile:157]: Hi good eye => we noticed this today. I think our checks missed that the previous hotfix PR was supposed to be updated to 1.42.1, but I accidentally updated the reference to 1.42.0 in final PR (I just added a comment on that PR with that info: https://github.com/wordpress-mobile/WordPress-iOS/pull/15436#issuecomment-743726518). As that hotfix was actually an android specific fix, so the PR was just an update to keep versioning in sync on the iOS side with no functional changes.

So yes the update from 1.42.0 1.42.2 is intentional.

**Review by cameronvoell — APPROVED** [review]
Verified crash fix and gb-mobile na dgutenberg hashes => ready to merge!

**Pull request merged**